corehq/apps/domain/views/settings.py

```diff
@@ -559,3 +569,19 @@ def post(self, request, *args, **kwargs):
         else:
             messages.error(request, _("There was an error saving your alert. Please try again!"))
             return self.get(request, *args, **kwargs)
+
+    def _update_alert(self, alert_id, command):
+        try:
+            alert = CommCareHQAlert.objects.get(
+                created_by_domain=self.domain,
+                id=alert_id
+            )
```

Comment:
If you were hypothetically to use `domains` instead of `created_by_domain`, you could fetch the instance like this:
```suggestion
            alert = CommCareHQAlert.objects.filter(
                domains__contains=[self.domain]
            ).get(id=alert_id)
```

Reply:
Replied on the other [comment](https://github.com/dimagi/commcare-hq/pull/33592#discussion_r1352322257) that created_by_domain is needed to correctly identify alerts created by the domain.